vl-convert-rs/src/converter.rs

```diff
@@ -218,24 +224,41 @@ import('{url}').then((sg) => {{
 
             // Create and initialize svg function string
             let function_str = r#"
-function vegaToView(vgSpec) {
+function vegaToView(vgSpec, allowedBaseUrls, errors) {
     let runtime = vega.parse(vgSpec);
-    const baseURL = 'https://vega.github.io/vega-datasets/';
+    let baseURL = 'https://vega.github.io/vega-datasets/';
     const loader = vega.loader({ mode: 'http', baseURL });
+    const originalHttp = loader.http.bind(loader);
+
+    if (allowedBaseUrls != null) {
+        loader.http = async (uri, options) => {
+            const parsedUri = new URL(uri);
+            if (
+                allowedBaseUrls.every(
+                    (allowedUrl) => !parsedUri.href.startsWith(allowedUrl),
+                )
+            ) {
+                errors.push(`External data url not allowed: ${uri}`);
+                throw new Error(`External data url not allowed: ${uri}`);
+            }
+            return originalHttp(uri, options);
+        };
+    }
+
     return new vega.View(runtime, {renderer: 'none', loader});
 }
 
-function vegaToSvg(vgSpec) {
-    let view = vegaToView(vgSpec);
+function vegaToSvg(vgSpec, allowedBaseUrls, errors) {
+    let view = vegaToView(vgSpec, allowedBaseUrls, errors);
     let svgPromise = view.toSVG().finally(() => { view.finalize() });
     return svgPromise
 }
 
-function vegaToScenegraph(vgSpec) {
-    let view = vegaToView(vgSpec);
+function vegaToScenegraph(vgSpec, allowedBaseUrls, errors) {
+    let view = vegaToView(vgSpec, allowedBaseUrls, errors);
     let scenegraphPromise = view.runAsync().then(() => {
         return JSON.parse(JSON.parse(vega.sceneToJSON(view.scenegraph())));
-    });
+    }).finally(() => { view.finalize() });
```

Comment:
Not related to this PR, but realized that we weren't finalizing the view when extracting the scenegraph